editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/preferences/PreferredLabelDialog.java

```diff
@@ -0,0 +1,210 @@
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.fusesource.ide.camel.editor.preferences;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.IInputValidator;
+import org.eclipse.jface.dialogs.IMessageProvider;
+import org.eclipse.jface.dialogs.TitleAreaDialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ModifyEvent;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+import org.fusesource.ide.camel.editor.internal.UIMessages;
+import org.fusesource.ide.camel.model.service.core.catalog.CamelModel;
+import org.fusesource.ide.camel.model.service.core.catalog.CamelModelFactory;
+import org.fusesource.ide.camel.model.service.core.catalog.Parameter;
+
+/**
+ * 
+ * @author Andrej Podhradsky (apodhrad@redhat.com)
+ *
+ */
+public class PreferredLabelDialog extends TitleAreaDialog {
+
+	private String component;
+	private String parameter;
+
+	private Combo componentCombo;
+	private Combo parameterCombo;
+
+	private IInputValidator componentValidator;
+	private IInputValidator parameterValidator;
+
+	public PreferredLabelDialog(Shell parentShell) {
+		super(parentShell);
+	}
+
+	public String getComponent() {
+		return component;
+	}
+
+	public void setComponent(String component) {
+		this.component = component;
+	}
+
+	public String getParameter() {
+		return parameter;
+	}
+
+	public void setParameter(String parameter) {
+		this.parameter = parameter;
+	}
+
+	public void setComponentValidator(IInputValidator componentValidator) {
+		this.componentValidator = componentValidator;
+	}
+
+	public void setParameterValidator(IInputValidator parameterValidator) {
+		this.parameterValidator = parameterValidator;
+	}
+
+	public void validate() {
+		if (componentValidator != null) {
+			if (setWarning(componentValidator.isValid(componentCombo.getText()))) {
+				return;
+			}
+		}
+		if (parameterValidator != null) {
+			setWarning(parameterValidator.isValid(parameterCombo.getText()));
+		}
+	}
+
+	public boolean setWarning(String msg) {
+		if ((msg == null || msg.isEmpty()) && getButton(IDialogConstants.OK_ID) != null) {
+			getButton(IDialogConstants.OK_ID).setEnabled(true);
+			setMessage(null, IMessageProvider.NONE);
+			return false;
+		}
+		setMessage(msg, IMessageProvider.WARNING);
+		getButton(IDialogConstants.OK_ID).setEnabled(false);
```

Comment:
5 lines above you check for getButton(IDialogConstants.OK_ID) != null but ,not (which explains my reported NPE)